src/SubsMgr.cpp

```diff
@@ -684,6 +687,19 @@ SubStatus TransactionSubsMgr::getFullStatus(const HashX &txHash) const
     return storage->getTxHeight(txHash);
 }
 
+ScriptHashTransactionsSubsMgr::~ScriptHashTransactionsSubsMgr() {}
+
+auto ScriptHashTransactionsSubsMgr::getFullStatus(const HashX &sh) const -> SubStatus
+{
+    auto [mempool, lock] = storage->mempool();
+    const auto shIt = mempool.scriptHashTransactionsAffected.find(sh);
```

Comment:
The new `scriptHashTransactionsAffected` is populated in `MempoolSynchTask` and lives only for one iteration of the task (for a single synch). This is then safe to assume, that this map of { Scripthash ; HashSet } contains only the new transactions.